qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -572,12 +573,15 @@ def update_toolbar_waterfall_plot(self, is_waterfall):
         self.toolbar.set_generate_plot_script_enabled(not is_waterfall)
 
     def change_line_collection_colour(self, colour):
-        for col in self.canvas.figure.get_axes()[0].collections:
-            if isinstance(col, LineCollection):
+        axes = self.canvas.figure.get_axes()[0]
+        ax_type = axes_type(axes)
+        if ax_type == FigureType.Wireframe:
+            for col in axes.collections:
                 col.set_color(colour.name())
-            elif isinstance(col, PathCollection):
-                col.set_edgecolor(colour.name())
-
+        elif ax_type == FigureType.Contour:
+            path_cols = self.canvas.figure.findobj(PathCollection)
```

Comment:
I really like this, I didn't know about `findobj` method. I feel there is a simpler way of rewriting this function using just this method.